epochStart/bootstrap/process.go

```diff
@@ -823,6 +823,11 @@ func (e *epochStartBootstrap) createTriesComponentsForShardId(shardId uint32) er
 		return err
 	}
 
+	err = e.tryCloseExisting(factory.UserAccountTrie)
```

Comment:
Cen we move the new code (L826-L830 + L844-L848) into a new function that will be called before we call `e.createTriesComponentsForShardId()` ? This is as to separate concerns about what the function should do. Or, in case we want that the `createTriesComponentsForShardId` should always close existing, call the new function on L802.

Reply:
i used the second suggestion, call the new closeFunctions inside the createTriesComponentsForShardId because this is going to overwrite the previous ones.